src/OpenTelemetry.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs

```diff
@@ -71,6 +80,8 @@ public override void OnStopActivity(Activity activity, object payload)
             try
             {
                 this.TransformMassTransitTags(activity);
+
+                this.options.Enrich?.Invoke(activity, "OnStopActivity", payload);
```

Comment:
Use this `MassTransitInstrumentationEventSource.Log.EnrichmentException("OnStopActivity", ex);` in the catch block?